Vagrantfile

```diff
@@ -24,53 +24,72 @@ Vagrant.configure("2") do |config|
 # Get the VM up-to-date
     sudo apt-get update
 
-# Install opam and things for compiling xapi
-    sudo apt-get install -y opam m4 libxen-dev git
+# Install opam and dependencies for compiling xapi
+    sudo apt-get install -y opam m4 libxen-dev
     opam init -a --compiler=4.02.3 -y
+    eval `opam config env`
     opam remote add xs-opam git://github.com/xapi-project/xs-opam
-    opam install -y merlin ocp-indent ocp-index ocp-browser utop lwt_react depext camlp4
+    opam install -y lwt_react depext camlp4
     sudo opam depext -y xapi
+    # due to constraints missing from old libraries, pinning the lwt package is necessary for certain configurations
     opam pin add lwt 2.7.1
     opam install --deps-only xapi
 
-# Install editors
-    sudo apt-get install xubuntu-desktop gksu leafpad synaptic
-    sudo add-apt-repository -y "ppa:webupd8team/sublime-text-3"
-    sudo apt-add-repository -y "ppa:webupd8team/atom"
```

Comment:
I think all the `add-apt-repository` were collected together so that we can issue one `apt-get update` instead of multiple ones, and save some time. However, in this new form, this file is more useful as a reference and easier to read and pick only the editors one needs, and also matches what is said in the README. So I think it's fine that we do `apt-get update` multiple times, which is now required because all the editor-specific commands are collected together.